lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/DriverTaskState.cs

```diff
@@ -0,0 +1,137 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Org.Apache.REEF.IMRU.OnREEF.Exceptions;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.IMRUTasks
+{
+    /// <summary>
+    /// Driver Task State represents task states from creating a new task, to submitted task, to task running, until task is completed. 
+    /// It also defines the state transition condition from one to another
+    /// All the task states are defined in <see cref="TaskState"></see>
+    /// For the task state transition diagram <see href="https://issues.apache.org/jira/browse/REEF-1327"></see>
+    /// This class wraps current state and provides methods to move from one state to the next state
+    /// </summary>
+    internal sealed class DriverTaskState
+    {
+        private readonly static IDictionary<TaskStateTransition, TaskState> Transitions = new ReadOnlyDictionary<TaskStateTransition, TaskState>(
+            new Dictionary<TaskStateTransition, TaskState>
+        {
+            { new TaskStateTransition(TaskState.TaskNew, TaskEvent.SubmittedTask), TaskState.TaskSubmitted },
+            { new TaskStateTransition(TaskState.TaskNew, TaskEvent.ClosedTask), TaskState.TaskClosedByDriver },            
+            { new TaskStateTransition(TaskState.TaskSubmitted, TaskEvent.RunningTask), TaskState.TaskRunning },
+            { new TaskStateTransition(TaskState.TaskSubmitted, TaskEvent.FailedTaskAppError), TaskState.TaskFailedAppError },
+            { new TaskStateTransition(TaskState.TaskSubmitted, TaskEvent.FailedTaskSystemError), TaskState.TaskFailedSystemError },
+            { new TaskStateTransition(TaskState.TaskSubmitted, TaskEvent.FailedTaskEvaluatorError), TaskState.TaskFailedByEvaluatorFailure },
+            { new TaskStateTransition(TaskState.TaskSubmitted, TaskEvent.FailedTaskCommunicationError), TaskState.TaskFailedByGroupCommunication },
+            { new TaskStateTransition(TaskState.TaskRunning, TaskEvent.CompletedTask), TaskState.TaskCompleted },
+            { new TaskStateTransition(TaskState.TaskRunning, TaskEvent.WaitingTaskToClose), TaskState.TaskWaitingForClose },
+            { new TaskStateTransition(TaskState.TaskRunning, TaskEvent.FailedTaskAppError), TaskState.TaskFailedAppError },
+            { new TaskStateTransition(TaskState.TaskRunning, TaskEvent.FailedTaskSystemError), TaskState.TaskFailedSystemError },
+            { new TaskStateTransition(TaskState.TaskRunning, TaskEvent.FailedTaskEvaluatorError), TaskState.TaskFailedByEvaluatorFailure },
+            { new TaskStateTransition(TaskState.TaskRunning, TaskEvent.FailedTaskCommunicationError), TaskState.TaskFailedByGroupCommunication },
+            { new TaskStateTransition(TaskState.TaskWaitingForClose, TaskEvent.ClosedTask), TaskState.TaskClosedByDriver },
```

Comment:
Where are the states here for failure events...didn't we agree to transition to TaskEvent.ClosedTask in failure cases if state is TaskState.TaskWaitingForClose